Java/TelemetrySystem/src/main/java/tddmicroexercises/telemetrysystem/TelemetryDiagnosticControls.java

```diff
@@ -3,42 +3,42 @@
 public class TelemetryDiagnosticControls
 {
     private final String DiagnosticChannelConnectionString = "*111#";
-    
-    private final TelemetryClient telemetryClient;
+    private Communication telemetryClient;
+    private Connection connection;
+    public void setTelemetryClient(Communication telemetryClient){
+        this.telemetryClient=telemetryClient;
+    }
+    public void setConnection(Connection connection) {
+        this.connection = connection;
+    }
     private String diagnosticInfo = "";
+    public String getDiagnosticInfo(){
+        return diagnosticInfo;
+    }
+    public void setDiagnosticInfo(String diagnosticInfo){
+
+        this.diagnosticInfo = diagnosticInfo;
+    }
+
+    public void checkTransmission() throws Exception

```

Comment:
segregate this method into an interface and implement it here

Reply:
updated